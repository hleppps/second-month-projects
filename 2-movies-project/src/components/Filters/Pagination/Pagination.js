import React from "react";
import { Button } from "react-bootstrap";
import * as CONSTANTS from "../../../utils/constants";
import styles from "./Pagination.module.scss";

const Pagination = (props) => {
  const onClickHandler = (changeType) => {
    props.setPageFilter(changeType, props.filteredPage);
  };

  return (
    <div className={styles.pagination}>
      <div className={styles.buttons}>
        <Button
          data-testid="decrease-button"
          variant="outline-secondary"
          className="m-1 w-100"
          onClick={() => {
            onClickHandler(CONSTANTS.PAGINATION_DECREASE);
          }}
        >
          «
        </Button>
        <Button
          data-testid="increase-button"
          variant="outline-secondary"
          className="m-1 w-100"
          onClick={() => {
            onClickHandler(CONSTANTS.PAGINATION_INCREASE);
          }}
        >
          »
        </Button>
      </div>
      <p className={styles.pages}>
        <span>Page:</span> {props.filteredPage} of {props.totalPages}
      </p>
    </div>
  );
};

export default Pagination;
