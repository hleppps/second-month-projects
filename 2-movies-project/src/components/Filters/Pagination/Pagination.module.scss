.pagination {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 1rem;
}

.buttons {
  width: 100%;
  display: flex;
}

.pages span {
  text-decoration: underline;
}