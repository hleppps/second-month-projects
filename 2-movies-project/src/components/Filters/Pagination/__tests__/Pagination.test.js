import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Pagination from "../Pagination";

afterEach(cleanup);

describe("Pagination component", () => {
  it("render page number from props", () => {
    const curPage = 1;
    const totalPages = 333;

    render(<Pagination filteredPage={curPage} totalPages={totalPages} />);

    const curPageOnScreen = screen.getByText(curPage, { exact: false });
    expect(curPageOnScreen).toBeInTheDocument();

    const totalPagesOnScreen = screen.getByText(totalPages, { exact: false });
    expect(totalPagesOnScreen).toBeInTheDocument();
  });

  it("calls the page change function on click", () => {
    const pageChangeHandler = jest.fn();

    const { getByTestId } = render(
      <Pagination setPageFilter={pageChangeHandler} />
    );

    fireEvent.click(getByTestId("decrease-button"));

    expect(pageChangeHandler).toBeCalled();

    fireEvent.click(getByTestId("increase-button"));

    expect(pageChangeHandler).toBeCalled();
  });

  it("matches a snapshot", () => {
    const { asFragment } = render(<Pagination />);
    expect(asFragment(<Pagination />)).toMatchSnapshot();
  });
});
