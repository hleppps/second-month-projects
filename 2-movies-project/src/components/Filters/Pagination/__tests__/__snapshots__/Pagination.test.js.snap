// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <div
      class="buttons"
    >
      <button
        class="m-1 w-100 btn btn-outline-secondary"
        data-testid="decrease-button"
        type="button"
      >
        «
      </button>
      <button
        class="m-1 w-100 btn btn-outline-secondary"
        data-testid="increase-button"
        type="button"
      >
        »
      </button>
    </div>
    <p
      class="pages"
    >
      <span>
        Page:
      </span>
        of 
    </p>
  </div>
</DocumentFragment>
`;
